geode-core/src/integrationTest/java/org/apache/geode/internal/cache/partitioned/ModifyColocationIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.partitioned;
+
+import static org.apache.geode.cache.RegionShortcut.PARTITION_PERSISTENT;
+import static org.apache.geode.distributed.ConfigurationProperties.LOCATORS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.TemporaryFolder;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.CacheFactory;
+import org.apache.geode.cache.PartitionAttributesFactory;
+import org.apache.geode.cache.RegionFactory;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.test.junit.categories.RegionsTest;
+
+@Category(RegionsTest.class)
+public class ModifyColocationIntegrationTest {
+
+  private InternalCache cache;
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @After
+  public void tearDown() {
+    closeCache();
+  }
+
+  /**
+   * Test that a user is not allowed to change the colocation of a PR with persistent data.
+   */
+  @Test
+  public void testModifyColocation() throws Exception {
+    // Create PRs where region3 is colocated with region1.
+    createColocatedPRs("region1");
+
+    pause();
+
+    // Close everything
+    closeCache();
+
+    // Restart colocated with "region2"
+    Throwable thrown = catchThrowable(() -> createColocatedPRs("region2"));
+    assertThat(thrown)
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessageContaining("Cannot change colocated-with to")
+        .hasMessageContaining(
+            "because there is persistent data with different colocation. Previous configured value is");
+
+    pause();
```

Comment:
TODO: Need to discuss the CancelExceptions that are thrown during Cache creation when these pause() calls are removed.